src/main/java/com/openlattice/hazelcast/HazelcastMap.kt

```diff
@@ -59,13 +59,17 @@ import com.openlattice.rhizome.hazelcast.DelegatedStringSet
 import com.openlattice.rhizome.hazelcast.DelegatedUUIDSet
 import com.openlattice.shuttle.Integration
 import com.openlattice.shuttle.IntegrationJob
-import com.openlattice.rhizome.service.QueueState
-import org.slf4j.LoggerFactory
 import java.util.*
 
 class HazelcastMap<K, V> internal constructor(val name: String) : TypedMapIdentifier<K, V> {
     private val checker = instanceChecker.checkInstance(name)
 
+    init {
+        if (checker.ok()) {
+            valuesCache.add(this)
```

Comment:
How come we use a list as a cache and not a map, where the keys are the names?

Reply:
To keep the same sorting guarantees as the previous enum. Probably not necessary though.